### PR TITLE
Create HOWTO-Update-an-API-repo-to-use-Fargate.md

### PR DESCRIPTION
**Pull request by DingoEatingFuzz**
Documenting all the rakes I stepped on today.

### CODE REVIEW

File: docs/HOWTO-Update-an-API-repo-to-use-Fargate.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+**Problem:** Due to some migrations and silent errors, rebuilding old repos may result in ECR errors, which will fail silently.
+**What to look for:** An error message in Travis, hidden under the `deploy.sh` collapsed section that mentions something about bad permissions or non-existent repo.
+**Solution:** Verify that the ENV VARS set on the Travis project match the VARS expects in the `/bin` scripts for the repo. Update ENV VARS accordingly. This may require breaking `DOCKER_REPO` into `DOCKER_REPO_NAMESPACE` and `DOCKER_REPO`
+
+### Getting services to restart in ECS
+
+**Problem:** Due to changes required for Fargate task definitions, the AWS user used by Travis for a repo may not have all the necessary permissions.
+**What to look for:** An error message in Travis, hidden under the `deploy.sh` collapsed section that mentions "AWS User <user> doesn't have required permission <permission>".
+**Solution:** Update the user in AWS to have the missing permission. It is likely `iam:PassRole` if anything.
```
Comment (MikeTheCanuck) [inline docs/HOWTO-Update-an-API-repo-to-use-Fargate.md]: Can you say more about what the reader can/should do here? i.e. is it a matter of contacting Michael Lange to get a specific user enhanced, or is this something others can do themselves?

File: docs/HOWTO-Update-an-API-repo-to-use-Fargate.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+## Confirming a Successful Migration
+
+You know that the migration is successful once a Travis build on the master branch results in
+
+1. A new pushed container to the ECR registry for the correct ECR repo
+2. A new task definition in ECS for the correct task
+3. A task restart in ECS resulting in the current running task for the correct service to be the latest task definition revision as seen in the Travis logs
+4. The newest tasks for the service in ECS in a running state.
+5. No error messages in CloudWatch for the newly running task.
+6. Passing health check log messages in CloudWatch for the newly running task.
```
Comment (MikeTheCanuck) [inline docs/HOWTO-Update-an-API-repo-to-use-Fargate.md]: I usually fire up a browser and request the https://service.civicpdx.org/{service_name} base page as well.

Comment (DingoEatingFuzz) [reply docs/HOWTO-Update-an-API-repo-to-use-Fargate.md]: Very good call.

File: docs/HOWTO-Update-an-API-repo-to-use-Fargate.md (hunk 1 above)
Comment (DingoEatingFuzz) [reply docs/HOWTO-Update-an-API-repo-to-use-Fargate.md]: Updated.

**Pull request merged**